builtin/providers/aws/resource_aws_ses_receipt_rule.go

```diff
@@ -0,0 +1,730 @@
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"position": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+				Set: func(v interface{}) int {
+					var buf bytes.Buffer
+					m := v.(map[string]interface{})
+					buf.WriteString(fmt.Sprintf("%s-", m["topic_arn"].(string)))
+					buf.WriteString(fmt.Sprintf("%d-", m["position"].(int)))
+
+					return hashcode.String(buf.String())
+				},
+			},
+
+			"stop_action": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"scope": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"topic_arn": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"position": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+				Set: func(v interface{}) int {
+					var buf bytes.Buffer
+					m := v.(map[string]interface{})
+					buf.WriteString(fmt.Sprintf("%s-", m["scope"].(string)))
+					buf.WriteString(fmt.Sprintf("%s-", m["topic_arn"].(string)))
+					buf.WriteString(fmt.Sprintf("%d-", m["position"].(int)))
+
+					return hashcode.String(buf.String())
+				},
+			},
+
+			"workmail_action": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"organization_arn": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"topic_arn": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"position": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+				Set: func(v interface{}) int {
+					var buf bytes.Buffer
+					m := v.(map[string]interface{})
+					buf.WriteString(fmt.Sprintf("%s-", m["organization_arn"].(string)))
+					buf.WriteString(fmt.Sprintf("%s-", m["topic_arn"].(string)))
+					buf.WriteString(fmt.Sprintf("%d-", m["position"].(int)))
+
+					return hashcode.String(buf.String())
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsSesReceiptRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sesConn
+
+	createOpts := &ses.CreateReceiptRuleInput{
+		Rule:        buildReceiptRule(d, meta),
+		RuleSetName: aws.String(d.Get("rule_set_name").(string)),
+	}
+
+	if v, ok := d.GetOk("after"); ok {
+		createOpts.After = aws.String(v.(string))
+	}
+
+	_, err := conn.CreateReceiptRule(createOpts)
+	if err != nil {
+		return fmt.Errorf("Error creating SES rule: %s", err)
+	}
+
+	d.SetId(d.Get("name").(string))
+
+	return resourceAwsSesReceiptRuleUpdate(d, meta)
+}
+
+func resourceAwsSesReceiptRuleUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sesConn
+
+	updateOpts := &ses.UpdateReceiptRuleInput{
```

Comment:
this looks like the `UpdateReceiptRule` will be called everytime a param changes. I.e. if we change `after`, the UpdateReceiptRule will be called and then SetReceiptRulePosition will be called. Is this ok? Shouldn't we wrap the UpdateReceiptRule in a check for a Change of some sort?